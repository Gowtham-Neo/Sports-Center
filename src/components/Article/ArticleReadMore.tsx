import React, { Fragment } from "react";
import { Article } from "../../context/Article/types";
import { Dialog, Transition } from "@headlessui/react";
import { GrClose } from "react-icons/gr";

interface ArticleReadMoreProps {
  selectedArticle: Article | null;
  isOpen: boolean;
  closeModel: () => void;
}

const ArticleReadMore: React.FC<ArticleReadMoreProps> = ({
  selectedArticle,
  isOpen,
  closeModel,
}) => {
  return (
    <Transition appear show={isOpen} as={Fragment}>
      <Dialog as="div" className="relative z-10" onClose={closeModel}>
        <Transition.Child
          as={Fragment}
          enter="ease-out duration-300"
          enterFrom="opacity-0"
          enterTo="opacity-100"
          leave="ease-in duration-200"
          leaveFrom="opacity-100"
          leaveTo="opacity-0"
        >
          <div className="fixed inset-0 bg-black bg-opacity-25" />
        </Transition.Child>
        <div className="fixed inset-0 overflow-y-auto">
          <div className="flex items-center justify-center min-h-full p-4 text-center">
            <Transition.Child
              as={Fragment}
              enter="ease-out duration-300"
              enterFrom="opacity-0 scale-95"
              enterTo="opacity-100 scale-100"
              leave="ease-in duration-200"
              leaveFrom="opacity-100 scale-100"
              leaveTo="opacity-0 scale-95"
            >
              <Dialog.Panel className="w-full max-w-lg p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-xl rounded-2xl">
                <div className="absolute pt-2 pr-2 text-xl font-semibold duration-150 right-1 top-1 hover:rotate-90">
                  <button type="button" onClick={closeModel}>
                    <GrClose />
                  </button>
                </div>
                <div className="flex items-start justify-between">
                  {selectedArticle && (
                    <>
                      <div className="flex items-start">
                        <div className="w-full md:mt-3 md:text-left">
                          <Dialog.Title
                            as="h3"
                            className="text-lg font-medium leading-6 text-gray-900"
                          >
                            {selectedArticle.title}
                          </Dialog.Title>
                          <hr/>
                          <div className="mt-2">
                            <div className="w-full ">
                              <img
                                className="w-4/6 mx-auto mt-4 mb-2 rounded-lg h-3/6"
                                src={selectedArticle.thumbnail}
                                alt={selectedArticle.title}
                              />
                            </div>
                            <hr/>
                            <p className="mt-4 text-gray-800 md:text-justify md:mx-4">
                              {selectedArticle.content}
                            </p>
                          </div>
                        </div>
                      </div>
                    </>
                  )}
                </div>
              </Dialog.Panel>
            </Transition.Child>
          </div>
        </div>
      </Dialog>
    </Transition>
  );
};

export default ArticleReadMore;
