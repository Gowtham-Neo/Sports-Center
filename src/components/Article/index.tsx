import React from "react";
import ArticleList from "./Articles";

const Article: React.FC = () => {
  return (
    <div className=" overflow-x-auto ">
      <ArticleList/>
    </div>
  );
};

export default Article;
