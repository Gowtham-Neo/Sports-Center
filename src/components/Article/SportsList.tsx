import React, { useState, useEffect } from "react";
import { API_ENDPOINT } from "../../config/constants";
import { Sport } from "../../context/Article/types";
import YourNews from "../Preferences";

interface SportsListProps {
  onSportClick: (selectedSport: string) => void;
  onYourNewsClick: (isYourNewsClicked: boolean) => void;
}

const SportsList: React.FC<SportsListProps> = ({
  onSportClick,
  onYourNewsClick,
}) => {
  const [sports, setSports] = useState<Sport[]>([]);
  const [selectedSport, setSelectedSport] = useState<string>("");
  const [isYourNewsClicked, setIsYourNewsClicked] = useState(false);
  const auth_token = localStorage.getItem("auth_token");

  useEffect(() => {
    const fetchSports = async () => {
      try {
        const response = await fetch(`${API_ENDPOINT}/sports`);
        if (!response.ok) {
          throw new Error("Error while fetching sports data");
        }
        const data = await response.json();
        setSports(data.sports);
      } catch (error) {
        console.error("Error fetching sports data:", error);
      }
    };

    fetchSports();
  }, []);

  if (!auth_token) {
    useEffect(() => {
      if (sports.length > 0) {
        onSportClick(sports[0].name);
      }
    }, [sports]);
  }

  useEffect(() => {
    onYourNewsClick(isYourNewsClicked);
    if (auth_token) {
      setSelectedSport("YourNews");
    } else {
      setSelectedSport("Basketball");
    }
  }, []);

  const handleSportClick = (selectedSport: string) => {
    onSportClick(selectedSport);
    setIsYourNewsClicked(false);
    setSelectedSport(selectedSport);
  };
  const handleYourNewsClick = () => {
    setSelectedSport("YourNews");
    setIsYourNewsClicked(true);
    onYourNewsClick(true);
  };

  return (
    <div className="overflow-x-auto rounded-lg bg-slate-800">
      <ul className="flex p-5 space-x-5">
        {auth_token && (
          <button
            className={`font-medium border px-3 py-2 rounded-md text-gray-400 border-red-700  active:text-white hover:text-white border-none ${
              selectedSport === "YourNews"
                ? " from-red-700 to-green-500 bg-gradient-to-r text-white"
                : ""
            }`}
            onClick={() => {
              setIsYourNewsClicked(true);
              handleYourNewsClick();
            }}
          >
            Your News
          </button>
        )}
        {sports.map((sport) => (
          <button
            key={sport.id}
            className={`font-medium  text-gray-400 border-red-700 active:text-white hover:text-white ${
              selectedSport === sport.name ? "text-white underline" : ""
            }`}
            onClick={() => handleSportClick(sport.name)}
          >
            {sport.name}
          </button>
        ))}
      </ul>
    </div>
  );
};

export default SportsList;
