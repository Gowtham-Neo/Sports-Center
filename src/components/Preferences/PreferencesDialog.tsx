import React, { Fragment, useState, useEffect } from "react";
import { Dialog, Transition } from "@headlessui/react";
import { Sport, Team } from "../../context/Article/types";
import { API_ENDPOINT } from "../../config/constants";
import { usePreferencesDispatch } from "../../context/Preferences/context";
import { fetchPreferencesList } from "../../context/Preferences/action";

interface PreferencesDialogProps {
  isOpen: boolean;
  closeModel: () => void;
}

const PreferencesDialog: React.FC<PreferencesDialogProps> = ({
  isOpen,
  closeModel,
}) => {
  const auth_token = localStorage.getItem("auth_token");
  const [sports, setSports] = useState<Sport[]>([]);
  const [teams, setTeams] = useState<Team[]>([]);
  const [selectedSports, setSelectedSports] = useState<string[]>([]);
  const [selectedTeams, setSelectedTeams] = useState<string[]>([]);
  const preferencesDispatch = usePreferencesDispatch();

  useEffect(() => {
    const fetchSports = async () => {
      try {
        const response = await fetch(`${API_ENDPOINT}/sports`);
        if (!response.ok) {
          throw new Error("Error while fetching sports data");
        }
        const data = await response.json();
        setSports(data.sports);
      } catch (error) {
        console.error("Error fetching sports data:", error);
      }
    };

    const fetchTeams = async () => {
      try {
        const response = await fetch(`${API_ENDPOINT}/teams`);
        if (!response.ok) {
          throw new Error("Error while fetching team data");
        }
        const data = await response.json();
        setTeams(data);
      } catch (error) {
        console.error("Error fetching team data:", error);
      }
    };

    fetchSports();
    fetchTeams();
  }, []);

  useEffect(() => {
    const storedSports = localStorage.getItem("selectedSports");
    const storedTeams = localStorage.getItem("selectedTeams");

    if (storedSports) {
      setSelectedSports(JSON.parse(storedSports));
    }

    if (storedTeams) {
      setSelectedTeams(JSON.parse(storedTeams));
    }
  }, []);

  const handleSportChange = (sportName: string) => {
    const updatedSports = [...selectedSports];
    if (updatedSports.includes(sportName)) {
      updatedSports.splice(updatedSports.indexOf(sportName), 1);
    } else {
      updatedSports.push(sportName);
    }
    setSelectedSports(updatedSports);
  };

  const handleTeamChange = (teamName: string) => {
    const updatedTeams = [...selectedTeams];
    if (updatedTeams.includes(teamName)) {
      updatedTeams.splice(updatedTeams.indexOf(teamName), 1);
    } else {
      updatedTeams.push(teamName);
    }
    setSelectedTeams(updatedTeams);
  };

  const handleSavePreferences = async () => {
    try {
      const response = await fetch(`${API_ENDPOINT}/user/preferences`, {
        method: "PATCH",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${auth_token}`,
        },
        body: JSON.stringify({
          preferences: { selectedSports, selectedTeams },
        }),
      });
      const updatedData = await response.json();
      fetchPreferencesList(preferencesDispatch);

      console.log(updatedData)

      if (!response.ok) {
        throw new Error("Error updating user preferences");
      }

      console.log("User preferences updated successfully");

      localStorage.setItem("selectedSports", JSON.stringify(selectedSports));
      localStorage.setItem("selectedTeams", JSON.stringify(selectedTeams));

      closeModel();
    } catch (error) {
      console.error("Error updating user preferences:", error);
    }
  };

  return (
    <Transition appear show={isOpen} as={Fragment}>
      <Dialog as="div" className="relative z-10" onClose={closeModel}>
        <Transition.Child
          as={Fragment}
          enter="ease-out duration-300"
          enterFrom="opacity-0"
          enterTo="opacity-100"
          leave="ease-in duration-200"
          leaveFrom="opacity-100"
          leaveTo="opacity-0"
        >
          <div className="fixed inset-0 bg-black bg-opacity-25" />
        </Transition.Child>
        <div className="fixed inset-0 overflow-y-auto">
          <div className="flex items-center justify-center min-h-full p-4 text-center">
            <Transition.Child
              as={Fragment}
              enter="ease-out duration-300"
              enterFrom="opacity-0 scale-95"
              enterTo="opacity-100 scale-100"
              leave="ease-in duration-200"
              leaveFrom="opacity-100 scale-100"
              leaveTo="opacity-0 scale-95"
            >
              <Dialog.Panel className="w-full max-w-lg p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-xl rounded-2xl">
                <div className="flex items-start justify-between">
                  <div className="w-full md:mt-3 md:text-left">
                    <Dialog.Title
                      as="h3"
                      className="text-xl font-semibold leading-6 text-gray-900"
                    >
                      Preferences
                    </Dialog.Title>
                    <hr />

                    <div className="grid grid-cols-3 gap-3 mt-4">
                      <div className="col-span-3">
                        <h4 className="mb-2 text-lg font-medium">
                          Favourite Sports:
                        </h4>
                        <hr />

                        <div className="grid grid-cols-3 gap-6 mt-4">
                          {sports.map((sport) => (
                            <div key={sport.id} className="flex items-center ">
                              <input
                                type="checkbox"
                                id={sport.name}
                                name={sport.name}
                                checked={selectedSports.includes(sport.name)}
                                onChange={() => handleSportChange(sport.name)}
                                className="mr-1 cursor-pointer"
                              />
                              <label
                                htmlFor={sport.name}
                                className="mr-4 cursor-pointer"
                              >
                                {sport.name}
                              </label>
                            </div>
                          ))}
                        </div>
                      </div>
                      <div className="col-span-3 mt-4">
                        <h4 className="mb-2 text-lg font-medium">
                          Favourite Teams:
                        </h4>
                        <hr />

                        <div className="grid grid-cols-3 gap-6 mt-4">
                          {teams.map((team) => (
                            <div key={team.id} className="flex items-center">
                              <input
                                type="checkbox"
                                id={team.name}
                                name={team.name}
                                checked={selectedTeams.includes(team.name)}
                                onChange={() => handleTeamChange(team.name)}
                                className="mr-2 cursor-pointer"
                              />
                              <label
                                htmlFor={team.name}
                                className="mr-4 cursor-pointer"
                              >
                                {team.name}
                              </label>
                            </div>
                          ))}
                        </div>
                      </div>

                      <div className="flex flex-row items-end justify-center gap-3">
                        <button
                          onClick={handleSavePreferences}
                          className="w-full px-4 py-2 text-white bg-blue-600 rounded-md hover:bg-blue-700"
                        >
                          Save
                        </button>
                        <button
                          onClick={closeModel}
                          className="w-full px-4 py-2 text-white bg-gray-700 rounded-md hover:bg-gray-800"
                        >
                          Close
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </Dialog.Panel>
            </Transition.Child>
          </div>
        </div>
      </Dialog>
    </Transition>
  );
};

export default PreferencesDialog;
