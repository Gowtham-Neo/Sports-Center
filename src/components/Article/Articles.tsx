import React, { useEffect, useState } from "react";
import {
    useArticlesDispatch,
    useArticlesState,
} from "../../context/Article/context";
import { fetchArticles } from "../../context/Article/action";
import SportsList from "./SportsList";
// import FavoriteComponent from "./FavoriteComponent";
// import ReadMoreDialog from "./ReadMoreDialog";

const ArticleList: React.FC = () => {
  const [selectedSport, setSelectedSport] = useState<string>();

  const articlesDispatch = useArticlesDispatch();
  const articlesState = useArticlesState();
  const { articles, isLoading } = articlesState;

  useEffect(() => {
    if (selectedSport) {
      fetchArticles(articlesDispatch);
    }
  }, [selectedSport, articlesDispatch]);

  const handleSportClick = (selectedSport: string) => {
    setSelectedSport(selectedSport);
  };

  return (
    
      <div className="flex flex-col">
        <SportsList onSportClick={handleSportClick} />
        <div className="w-full">
          {isLoading ? (
            <p className="pt-10 font-serif text-xl ps-96">Loading...</p>
          ) : (
            <div>
              {articles
                .filter((article: any) =>
                  selectedSport
                    ? article.sport.name === selectedSport
                    : true
                )
                .map((article: any) => (
                  <div
                    key={article.id}
                    className="flex flex-col p-4 my-4 mb-4 bg-white border border-gray-300 rounded-lg shadow-md md:flex-row"
                  >
                    <div className="w-full md:w-3/4">
                      <h2 className="text-lg font-medium">{article.title}</h2>
                      <h2 className="text-gray-600">{article.summary}</h2>
                      <button
                        className="flex text-center text-gray-700 cursor-pointer underline-none hover:underline hover:text-gray-900"
                      >
                        Read More...
                      </button>
                      <p className="mt-2 text-gray-600">
                        {new Date(article.date).toLocaleDateString()}
                      </p>
                    </div>
                    <div className="w-full mt-4 md:w-1/4 md:mt-0">
                      <img
                        src={article.thumbnail}
                        alt={article.title}
                        className="shadow-lg rounded-xl shadow-black"
                      />
                    </div>
                  </div>
                ))}
            </div>
          )}
        </div>
      </div>
  );
};

export default ArticleList;