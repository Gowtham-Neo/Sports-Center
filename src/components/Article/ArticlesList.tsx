import React, { useEffect, useState } from "react";
import {
  useArticlesDispatch,
  useArticlesState,
} from "../../context/Article/context";
import { fetchArticles } from "../../context/Article/action";
import SportsList from "./SportsList";
import { usePreferencesDispatch } from "../../context/Preferences/context";
import { fetchPreferencesList } from "../../context/Preferences/action";
import YourNews from "../Preferences";
import { API_ENDPOINT } from "../../config/constants";

interface ArticleListProps {
  onArticleClick: (article: any) => void;
}

const ArticleList: React.FC<ArticleListProps> = ({ onArticleClick }) => {
  const [selectedSport, setSelectedSport] = useState<string | undefined>();
  const dispatch = useArticlesDispatch();
  const preferencesDispatch = usePreferencesDispatch();
  const { articles, isLoading, isError } = useArticlesState();

  useEffect(() => {
    if (selectedSport && selectedSport !== "Your News") {
      fetchArticles(dispatch);
    } else if (selectedSport === "Your News") {
      fetchPreferencesList(preferencesDispatch);
    }
  }, [selectedSport, dispatch, preferencesDispatch]);

  const handleSportClick = (selectedSport: string) => {
    setSelectedSport(selectedSport);
  };
  
  const handleArticleReadMore = async (articleId: number) => {
    try {
      const response = await fetch(`${API_ENDPOINT}/articles/${articleId}`, {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
        },
      });
      if (!response.ok) {
        throw new Error("Error while Fetching");
      }
      const data = await response.json();
      onArticleClick(data);
    

    } catch (error) {
      console.error("Error fetching article content:", error);
    }
  };


  return (
    <div className="flex flex-col">
      <SportsList
        onSportClick={handleSportClick}
        onYourNewsClick={() => setSelectedSport("Your News")}
      />
      <div className="w-full">
        {isLoading && (
          <p className="pt-10 font-serif text-xl ps-96">Loading...</p>
        )}
        {isError && (
          <p className="pt-10 font-serif text-xl ps-96">
            Error fetching Articles
          </p>
        )}
        {!isLoading && !isError && (
          <div>
            {selectedSport !== "Your News"  ? (
              articles
                .filter((article: any) =>
                  article.sport.name === selectedSport
                )
                .map((article: any) => (
                  <div
                    key={article.id}
                    className="flex flex-col p-4 my-4 mb-4 bg-white border border-gray-300 rounded-lg shadow-md md:flex-row"
                  >
                    <div className="w-full md:w-3/4">
                      <h2 className="text-lg font-medium">{article.title}</h2>
                      <h2 className="text-gray-600">{article.summary}</h2>
                      <button
                        onClick={() => handleArticleReadMore(article.id)}
                        className="flex text-center text-gray-700 cursor-pointer underline-none hover:underline hover:text-gray-900"
                      >
                        Read More...
                      </button>
                      <p className="mt-2 text-gray-600">
                        {new Date(article.date).toDateString()}
                      </p>
                    </div>
                    <div className="w-full mt-4 md:w-1/4 md:mt-0">
                      <img
                        src={article.thumbnail}
                        alt={article.title}
                        className="shadow-lg rounded-xl shadow-black"
                      />
                    </div>
                  </div>
                ))
            ) : (
               <YourNews />
            )}
          </div>
        )}
      </div>
    </div>
  );
};

export default ArticleList;
