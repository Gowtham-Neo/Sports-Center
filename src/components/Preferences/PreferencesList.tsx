import React, { useState, useEffect } from "react";
import { usePreferencesState } from "../../context/Preferences/context";
import { API_ENDPOINT } from "../../config/constants";

interface PreferencesListProps {
  onArticleClick: (article: any) => void;
}

const PreferencesList: React.FC<PreferencesListProps> = ({
  onArticleClick,
}) => {
  const { yourNews, isLoading, isError } = usePreferencesState();
  const auth_token = localStorage.getItem("auth_token");
  const [isEmpty, setEmpty] = useState(false);
  useEffect(() => {
    setEmpty(!yourNews || yourNews.length === 0);
  }, [yourNews]);

  const handleArticleReadMore = async (articleId: number) => {
    try {
      const response = await fetch(`${API_ENDPOINT}/articles/${articleId}`, {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${auth_token}`,
        },
      });
      if (!response.ok) {
        throw new Error("Error while Fetching");
      }
      const data = await response.json();
      onArticleClick(data);
    } catch (error) {
      console.error("Error fetching article content:", error);
    }
  };

  return (
    <div className="flex flex-col">
      <div className="w-full">
        {isLoading && (
          <p className="p-16 font-serif text-xl md:ps-80">Loading...</p>
        )}
        {isError && (
          <p className="p-16 font-serif text-xl md:ps-80">
            Error fetching Preferences
          </p>
        )}
        {!isLoading && !isError && (
          <div>
            {isEmpty ? (
              <h2 className="p-16 font-serif text-xl md:ps-80">
                No preferences Selected
              </h2>
            ) : (
              yourNews.map((article: any) => (
                <div
                  key={article.id}
                  className="flex flex-col p-4 my-4 mb-4 bg-white border border-gray-300 rounded-lg shadow-md md:flex-row"
                >
                  <div className="w-full md:w-3/4">
                    <h2 className="text-lg font-medium">{article.title}</h2>
                    <h2 className="text-gray-600">{article.summary}</h2>
                    <button
                      onClick={() => handleArticleReadMore(article.id)}
                      className="flex text-center text-gray-700 cursor-pointer underline-none hover:underline hover:text-gray-900"
                    >
                      Read More...
                    </button>
                    <p className="mt-2 text-gray-600">
                      {new Date(article.date).toDateString()}
                    </p>
                  </div>
                  <div className="w-full mt-4 md:w-1/4 md:mt-0">
                    <img
                      src={article.thumbnail}
                      alt={article.title}
                      className="shadow-lg rounded-xl shadow-black"
                    />
                  </div>
                </div>
              ))
            )}
          </div>
        )}
      </div>
    </div>
  );
};

export default PreferencesList;
