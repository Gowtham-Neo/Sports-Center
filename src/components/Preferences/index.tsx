import React, { useState } from "react";
import PreferencesList from "./PreferencesList";
import ArticleReadMore from "../Article/ArticleReadMore";
import { Article } from "../../context/Article/types";

const YourNews: React.FC = () => {
  const [selectedArticle, setSelectedArticle] = useState<Article | null>(null);

  const handleArticleClick = (article: Article) => {
    setSelectedArticle(article);
  };

  const handleCloseReadMore = () => {
    setSelectedArticle(null);
  };

  return (
    <div>
      <div className="overflow-x-auto">
        (
        <PreferencesList
          onArticleClick={handleArticleClick}
        />
        )
        <ArticleReadMore
          selectedArticle={selectedArticle}
          isOpen={selectedArticle !== null}
          closeModel={handleCloseReadMore}
        />
      </div>
    </div>
  );
};

export default YourNews;
