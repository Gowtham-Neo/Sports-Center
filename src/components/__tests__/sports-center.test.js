import React from 'react';
import { render, waitFor, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import ArticleList from '../Article/ArticlesList';
import { useArticlesDispatch, useArticlesState } from '../../context/Article/context';
import { usePreferencesDispatch } from '../../context/Preferences/context';


jest.mock('../../context/Article/context', () => ({
  useArticlesDispatch: jest.fn(),
  useArticlesState: jest.fn(),
}));

jest.mock('../../context/Preferences/context', () => ({
  usePreferencesDispatch: jest.fn(),
}));

jest.mock('../../context/Article/action', () => ({
  fetchArticles: jest.fn(),
}));

jest.mock('../../context/Preferences/action', () => ({
  fetchPreferencesList: jest.fn(),
}));

jest.mock('../Article/SportsList', () => () => <div>SportsList Component</div>);
jest.mock('../Preferences', () => () => <div>YourNews Component</div>);

describe('ArticleList', () => {
  let mockArticlesDispatch;
  let mockPreferencesDispatch;
  let mockOnArticleClick;

  beforeEach(() => {
    mockArticlesDispatch = jest.fn();
    mockPreferencesDispatch = jest.fn();
    mockOnArticleClick = jest.fn();

    useArticlesDispatch.mockReturnValue(mockArticlesDispatch);
    usePreferencesDispatch.mockReturnValue(mockPreferencesDispatch);
    useArticlesState.mockReturnValue({
      articles: [],
      isLoading: false,
      isError: false,
    });
  });

  afterEach(() => {
    cleanup();
  });

  it('Renders correctly', () => {
    const { container } = render(<ArticleList onArticleClick={mockOnArticleClick} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('Displays Loading State', () => {
    useArticlesState.mockReturnValue({
      articles: [],
      isLoading: true,
      isError: false,
    });

    const { getByText } = render(<ArticleList onArticleClick={mockOnArticleClick} />);
    expect(getByText('Loading...')).toBeInTheDocument();
  });

  it('Displays Error State', () => {
    useArticlesState.mockReturnValue({
      articles: [],
      isLoading: false,
      isError: true,
    });

    const { getByText } = render(<ArticleList onArticleClick={mockOnArticleClick} />);
    expect(getByText('Error fetching Articles')).toBeInTheDocument();
  });
});
