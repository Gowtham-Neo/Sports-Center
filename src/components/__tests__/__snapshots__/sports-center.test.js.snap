// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList Renders correctly 1`] = `"<div class="flex flex-col"><div>SportsList Component</div><div class="w-full"><div></div></div></div>"`;
